npm/test/folderinforetrieval.test.ts

```diff
@@ -57,60 +80,299 @@ const mockFormHTML = `
+    ...this.dataTransferItemFields,
+    createReader: () => this.reader,
+    type: "pdf", // overwrite default "type" value "" as files must have a non-empty value
+    isFile: true,
+    isDirectory: false,
+    webkitGetAsEntry: jest.fn()
+  }
+  directoryEntry: IWebkitEntry = {
+    ...this.dataTransferItemFields,
+    createReader: () => this.reader,
+    isFile: false,
+    isDirectory: true,
+    webkitGetAsEntry: () => this.fileEntry
+  }
+
+  dataTransferItem: DataTransferItem = {
+    ...this.dataTransferItemFields,
+    webkitGetAsEntry: () => this.directoryEntry
+  }
+
+  entries: IWebkitEntry[][] = [[], [this.fileEntry, this.fileEntry]]
+  batchCount = this.entries.length
+  reader: IReader = {
+    readEntries: (cb) => {
+      this.batchCount = this.batchCount - 1
+      cb(this.entries[this.batchCount])
+    }
+  }
+
+  mockFileList: (file: File[]) => FileList = (file: File[]) => {
+    return {
+      length: file.length,
+      item: (index: number) => {
+        return file[index]
+      }
+    } as FileList
+  }
+
+  mockDataTransferItemList: (
+    entry: DataTransferItem,
+    itemLength: number
+  ) => DataTransferItemList = (entry: DataTransferItem, itemLength: number) => {
+    return {
+      item: jest.fn(),
+      [Symbol.iterator]: jest.fn(),
+      add: jest.fn(),
+      length: itemLength,
+      clear: jest.fn(),
+      0: entry,
+      remove: jest.fn()
+    } as DataTransferItemList
+  }
+
+  addFilesToDragEvent = (
+    folderToDrop: File[],
+    itemsToDropEntryType: DataTransferItem
+  ) => {
+    return class MockDragEvent extends MouseEvent {
+      constructor() {
+        super("drag")
+      }
+      dataTransfer = {
+        files: mockFileList(folderToDrop),
+        dropEffect: "",
+        effectAllowed: "",
+        items: mockDataTransferItemList(
+          itemsToDropEntryType,
+          folderToDrop.length
+        ),
+        types: [],
+        clearData: jest.fn(),
+        getData: jest.fn(),
+        setData: jest.fn(),
+        setDragImage: jest.fn()
+      }
+    }
+  }
+
+  dropzone: HTMLElement | null = document.querySelector(
+    ".drag-and-drop__dropzone"
+  )
+  uploadForm: HTMLFormElement | null = document.querySelector(
+    "#file-upload-form"
+  )
+  folderRetriever: HTMLInputElement | null = document.querySelector(
+    "#file-selection"
+  )
+
+  folderRetrievalSuccessMessage: HTMLElement | null = document.querySelector(
+    ".drag-and-drop__success"
+  )
+
+  folderRetrievalfailureMessage: HTMLElement | null = document.querySelector(
+    ".drag-and-drop__failure"
+  )
+  folderNameElement: HTMLElement | null = document.querySelector("#folder-name")
+  folderSizeElement: HTMLElement | null = document.querySelector("#folder-size")
+
+  form = new UploadForm(this.uploadForm!, this.folderRetriever!, this.dropzone!)
+  fileUploader = this.setUpFileUploader()
+
+  setUpFileUploader(): FileUploader {
+    const client = new GraphqlClient("https://test.im", mockKeycloakInstance)
```

Comment:
If this is an arbitrary URL, you normally want to use example.com because it was set up be IETF for that purpose: https://en.wikipedia.org/wiki/Example.com

Reply:
It's arbitrary, I've updated to use `example.com`

Reply:
It looks like it's still test.im. Is there a commit that hasn't been pushed?

Reply:
Not sure what happened there. I've updated it again.